### ISSUE TITLE
[BUG] broken /usr/lib64/cmake/OpenImageIO/OpenImageIOConfig.cmake generated during build

### ISSUE THREAD

**Issue by darix** [AcademySoftwareFoundation/OpenImageIO#4049]
**Describe the bug**

While investigating why our openSUSE blender package does not build anymore i tracked this down to

```
# add here all the find_dependency() whenever switching to config based dependencies
if (NOT OFF AND NOT OPENIMAGEIO_CONFIG_DO_NOT_FIND_IMATH)
    if (3.1.11 VERSION_GREATER_EQUAL 3.0)
        find_dependency(Imath 3.1.9
                        HINTS /usr/lib64/cmake/Imath)
    elseif (3.1.11 VERSION_GREATER_EQUAL 2.4 AND 1)
        find_dependency(IlmBase 3.1.11
                        HINTS  /usr/lib64/cmake/OpenEXR)
        find_dependency(OpenEXR 3.1.11
                        HINTS /usr/lib64/cmake/OpenEXR)
        find_dependency(ZLIB 1.3.#define ZLIB_VERSION "1.3")  # Because OpenEXR doesn't do it
        find_dependency(Threads)  # Because OpenEXR doesn't do it
    endif ()
endif ()
```

This part looks really broken:

```
  find_dependency(ZLIB 1.3.#define ZLIB_VERSION "1.3")  # Because OpenEXR doesn't do it
```

**To Reproduce**

openSUSE package at https://build.opensuse.org/package/show/home:darix:branches:graphics/OpenImageIO

**Expected behavior**

a correct file being generated

**Platform information:**
 - OIIO branch/version: 2.5.5.0 and 2.4.14.0
 - OS: openSUSE Tumbleweed
 - C++ compiler: gcc13-c++-13.2.1+git7813-2.2.x86_64
 - Any non-default build flags when you build OIIO: see build log
 -  cmake version: cmake-3.27.2-1.1.x86_64

**Comment by jessey-git** [comment]
I'm not able to see your build logs. Says "Package not found: home:darix:branches:graphics/OpenImageIO"

Also, can you confirm that things used to work when using zlib 1.2.x? zlib 1.3 is somewhat new so your previous packages were probably using 1.2 in some form.

**Comment by darix** [comment]
you can find the package here now:

https://build.opensuse.org/package/show/graphics/OpenImageIO

as you can see in the spec file I added a workaround with a little perl search and replace to workaround the problem.
yes we used to have zlib 1.2. but openSUSE Tumbleweed is adopting new versions quite quickly. so we switched to zlib 1.3 a few weeks ago. which broke a few packages. e.g. some had assumptions about triplets as version numbers and well 1.3 does not have that :)

**Comment by darix** [comment]
so i checked the 15.5 build which still uses zlib 1.2.x

```
find_dependency(ZLIB 1.2.13)  # Because OpenEXR doesn't do it
```

that looks correct. but this also uses an older cmake. so not sure how comparable that is.

**Comment by darix** [comment]
JFYI just found another package which triggers the same problem

```
[    5s] -- Found ZLIB: /usr/lib64/libz.so (found version "1.3.#define ZLIB_VERSION "1.3"")  
```

**Comment by darix** [comment]
https://github.com/Kitware/CMake/commit/903439495033da40f798db9ec064d9b16ea672fc

**Comment by darix** [comment]
this can be closed as it was a generic cmake bug in 3.27.2 which is fixed in 3.27.7

**Comment by jessey-git** [comment]
Good find. Thank you for that followup!
